cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -955,8 +957,8 @@ public void liveInfo(HttpRequest request, HttpResponder responder, @PathParam("n
                        @PathParam("program-id") String programId) {
     ProgramType type = getProgramType(programCategory);
     if (type == null) {
-      responder.sendString(HttpResponseStatus.METHOD_NOT_ALLOWED, String.format("Live-info not supported for program" +
-                                                                                  " type '%s'", programCategory));
+      responder.sendString(HttpResponseStatus.BAD_REQUEST,
```

Comment:
I think it's straightforward, but I probably used the wrong HTTP code here.

`Method Not Allowed` is for when you make a request to a path that doesn't support the HTTP method in your request. In this case, `Not Found` would probably make the most sense since conceptually there isn't a live-info endpoint for program types that don't support it.